packages/tfjs-node-helpers/src/testing/metrics/pr-auc.ts

```diff
@@ -0,0 +1,40 @@
+import { binarize } from '../../utils/binarize';
+import { Metric } from '../metric';
+import { MetricCalculator, MetricCalculatorData } from '../metric-calculator';
+import { PrecisionMetricCalculator } from './precision';
+import { RecallMetricCalculator } from './recall';
+
+export class PRAucMetricCalculator extends MetricCalculator {
+  constructor(private thresholdNumber: number = 21) {
+    super();
+  }
+
+  public calculate({ trueValues, probabilities }: MetricCalculatorData): Metric {
+    let area = 0;
+    let threshold = 0;
+    const step = 1 / (this.thresholdNumber - 1);
+    const precisions: Array<number> = [];
+    const recalls: Array<number> = [];
+    const precisionCalculator = new PrecisionMetricCalculator();
+    const recallCalculator = new RecallMetricCalculator();
+
+    for (let i = 0; i < this.thresholdNumber; i++) {
+      const predictedValues = binarize(probabilities, threshold);
+
+      precisions.push(precisionCalculator.calculate({ trueValues, probabilities, predictedValues }).value);
+      recalls.push(recallCalculator.calculate({ trueValues, probabilities, predictedValues }).value);
+
+      if (i > 0) {
+        area += ((precisions[i] + precisions[i - 1]) * (recalls[i - 1] - recalls[i])) / 2;
+      }
+
+      threshold += step;
+      predictedValues.dispose();
```

Comment:
@VladimirGlinskii It looks like we could dispose `predictedValues` right after pushing the calculated `recall`.